src/mgmt/srv_pool.c

```diff
@@ -668,8 +716,42 @@ ds_mgmt_pool_target_update_state(uuid_t pool_uuid, uint32_t rank,
 {
 	int			rc;
 	d_rank_list_t		*ranks;
+	d_rank_list_t		*reint_ranks = NULL;
 	struct mgmt_svc		*svc;
 
+	if (state == PO_COMP_ST_UP) {
+		/* When doing reintegration, need to make sure the pool is
+		 * created and started on the target rank
+		 */
+
+		reint_ranks = d_rank_list_alloc(1);
+		if (reint_ranks == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+
+		reint_ranks->rl_nr = 1;
+		reint_ranks->rl_ranks[0] = rank;
+
+		/* TODO: The size information and "pmem" type need to be
+		 * determined automatically, perhaps by querying the pool leader
+		 * This works for now because these parameters are ignored if
+		 * the pool already exists on the destination node. This is
+		 * just used to ensure the pool is started.
+		 *
+		 * Fixing this will add the ability to reintegrate with a new
+		 * node, rather than only the previously failed node.
+		 *
+		 * This is tracked in DAOS-5041
+		 */
+		rc = ds_mgmt_tgt_pool_create_ranks(pool_uuid, "pmem",
+						   reint_ranks, 0, 0, NULL);
```

Comment:
what is this "pmem" for? it seems tc_tgt_dev is not being used anymore. i.e. you do not need set tc_tgt_dev.

Reply:
I assume you need query the status of the pool on the new rank, and do some validation check, then decide whether delete and recreate  or create or  use the current one? 

 Probably those container deletion code in migration should be moved here,  then you can test pool creation in this process, and also be easier to remove those temporary code later.

Reply:
pmem is not used. I'm going to create a separate patch to remove this argument. Since landing this quickly is a priority, I don't want to include it here. 

Agreed on moving the container deletion code here, I'll look at doing that also in a separate patch.